### ISSUE TITLE
Room list filter UI should give you feedback if the all_rooms list is not up-to-date.

### ISSUE THREAD

**Issue by ara4n** [element-hq/element-x-ios#1203]
### Steps to reproduce

1. Launch the app after being offline for a while (so SS has expired)
2. Immediately try to search for a room
3. Room may not be returned in the room list, because the all_rooms SS list has yet to update.
4. The user has no idea that the room list search results are incomplete

Therefore, we should show some kind of UI to warn the user if the room list search results are partial, so they know to wait until the all-rooms list has updated - e.g. a spinner row in the results.

#### What did you expect?

To be warned if room list filter results are partial

#### What happened instead?

No warning; just rooms failed to turn up until i waited a few seconds and tried again.

### Application version

257

### Will you send logs?

No

**Comment by ara4n** [comment]
this isn't that uncommon - it happens for every user every time they foreground the app after being offline for >30 minutes, and will affect any room searches done until all_rooms has synced up again.  (With the room cache you'll get stale results returned at least).  It will also happen 100% for every user on a fresh login.

**Comment by ara4n** [comment]
this bug still exists, and hasn't been fixed, and still causes problems: for instance, right now, we have anecdotal reports of room search not returning room items. We don't know if this is because the roomlist is stale, or if the app thinks the roomlist is up-to-date and something else has gone wrong.

In other words, we still need some mechanism which tells you, when you search, if it's searching incomplete state or not. The normal idiom would be to put a spinner in the search results to let you know it's showing you an incomplete search result.

**Comment by ara4n** [comment]
I just had a situation where search silently failed to return an older room i *knew* existed. I ended up waiting about 5 minutes and then trying again, and this time it worked.  The lack of UX feedback is 😬

**Comment by kongo09** [comment]
I just hit the same problem on Android

![image](https://github.com/user-attachments/assets/0e877bf3-9dfe-40c7-8cb3-ac68a391c7e1)

However, I don't think as a user I would expect feedback that the room list in not up-to-date. This is a technical state that an ordinary user cannot do anything with and wouldn't even understand (what is a room list in the first place?)

The effect shouldn't even exist in such a drastic way. With rooms and the room list being cached, at least the case I've captured in the screen above wouldn't exist. The room I'm searching for is not new, has not a changed name, is frequented by me regularly - there is just no good reason why it wouldn't show. If the app now told me _"my room list is incomplete"_ I'd be even more disappointed as it just feels like broken by desgin.

**Comment by ara4n** [comment]
> The effect shouldn't even exist in such a drastic way

Agreed, but I believe we're stuck with the current implementation until either persistent event cache lands (roughly end of year) or paginated incremental sync lands in SSS (not even scheduled currently).  Instead, for now, the app re-paginates in its roomlist whenever it's been offline for 30 mins - and so for a few seconds after launch it'll not be able to find old rooms in search.  So, i maintain that we need to provide a spinner to warn the user to wait a few more seconds if they want reliable search results, for this scenario - even though this scenario will get less frequent in future.

**Comment by Hywan** [comment]
Duplicate of https://github.com/element-hq/element-x-ios/issues/3365 I suppose.
